Core/tests/ExpressionParser2.cpp

```diff
@@ -573,22 +673,36 @@ TEST_CASE("ExpressionParser2", "[common][events]") {
 
   SECTION("Invalid number operators") {
     {
-      auto node = parser.ParseExpression("number", "123 % 456");
+      auto node = parser.ParseExpression("123 % 456");
       REQUIRE(node != nullptr);
 
-      gd::ExpressionValidator validator;
+      gd::ExpressionValidator validator(platform, project, layout1, "number");
       node->Visit(validator);
-      REQUIRE(validator.GetErrors().size() == 2);
-      REQUIRE(validator.GetErrors()[0]->GetMessage() ==
+      REQUIRE(validator.GetErrors().size() == 3);
+      REQUIRE(validator.GetErrors()[1]->GetMessage() ==
               "No operator found. Did you forget to enter an operator (like +, "
               "-, * or /) between numbers or expressions?");
-      REQUIRE(validator.GetErrors()[0]->GetStartPosition() == 4);
+      REQUIRE(validator.GetErrors()[1]->GetStartPosition() == 4);
+      // TODO Should error be removed?
+      REQUIRE(validator.GetErrors()[2]->GetMessage() ==
+              "You must enter a number.");
+    }
```

Comment:
Here additional errors happen because "%" is not a reserved character and it's read as an identifier. This PR adds a second error that is not really pertinent, but I don't see an easy solution to remove it. Should it be dug?

The test that follows use a reserved character and only the pertinent message is shown. I guess, that operators commonly used in other language can't be added to the list of reserved character because of side effects.

Reply:
>  I guess, that operators commonly used in other language can't be added to the list of reserved character because of side effects.

Maybe we could handle "well known" characters like % to ensure we deliver a proper error message (like, "the modulo operator is not supported, using `mod` instead"). Not a big priority though, it's rather a "nice to have".